### PR TITLE
Resolve conflit between Tchap and Element Web 1.11.25

### PR DESCRIPTION
**Pull request by jdauphant**
# ToDo

- [x] resolv conflict
- [ ] Fix LintError in Tchap File
- [x] resolve patches problem
  - [x] Remove green-confirmation-shield
  - [x] update patches with no conflits
  - [x] update other patches

Patches migration : 
- [x] activate-cross-signing-and-secure-storage-js (automatic)
- [x] activate-cross-signing-and-secure-storage-react (olivier)
- [ ] TEST : device verified without xs
- [x] activate-expired-account-panel (automatic)
- [x] add-a-help-tab-in-menu-to-redirect-to-external-tchap-faq  (olivier)
- [x] TEST : help link  in menu
- [x] are-external-users-allowed-in-room (automatic)
- [x] auto-accept-tac  (olivier)
- [ ] TEST : terms are auto accepted 
- [x] better-help-settings  (olivier)
- [x] TEST : help settings are visually correct
- [x] better-help-settings-2 (automatic)
- [x] content-scanner (automatic)
- [x] cross-signing-ui  (olivier)
- [x] TEST : relogin screen without xsss
- [x] disable-access-options  (olivier)
- [x] fix-inviting-a-person-already-present-in-the-room (automatic)
- [x] forgot-password (olivier)
- [x] TEST : test forgot password
- [x] hide-discovery-email-phone-settings (automatic)
- [x] hide-location-if-no-map-server (julien)
- [x] hide-room-alias-settings (automatic)
- [x] in-invite-dialog-add-text-indicating-that-we-can-invite-multiple-people (automatic)
- [x] login (olivier)
- [x] password-policy (olivier)
- [x] public-room-server-list (automatic)
- [x] registration-for-mainlining (automatic)
- [x] show-icon-on-public-room (automatic)
- [x] simplify-exchange-key-message (julien)
- [x] temp-settings-remove-enable-email-notifications-option (automatic)
- [x] use-the-term-people-not-direct-messages (automatic)
- [x] user-friendlier-encrypted-messages (removed by Julien)
- [x] ux-improvements-for-xsss (automatic)

# Notes

- [.github/deactivated_workflows/triage-priority-bugs.yml](https://github.com/tchapgouv/tchap-web-v4/pull/475/files#diff-95d0db1fa1b2780309c33c8aaa7c89f3ca0461cf768dd237178b41a33ef96297) have been removed upstream

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -159,11 +78,23 @@
         "matrix-js-sdk": "23.5.0",
         "matrix-react-sdk": "3.68.0",
         "matrix-widget-api": "^1.1.1",
+        "patch-package": "^6.5.1",
```
Comment (jdauphant) [inline package.json:81]: I have added it to the wrong section

File: scripts/merge-patches.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+  if [ "$CONFLICTS_FOUND" = false ]; then
+    echo "All patches have been updated without conflicts."
+    clean_temp_dir
+  else
+    echo "Some patches have conflicts. After resolving the conflicts, run the script again with the 'continue' command."
+  fi
+}
+
+function continue_patches() {
```
Comment (odelcroi) [inline scripts/merge-patches.sh:83]: what is the difference between 'merge' and 'continue' procedure? I don't see it

File: scripts/merge-patches.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    LOG_FILE="$PACKAGE_TEMP_DIR".log
+
+    # Install the package in the package subfolder
+    cd "$PACKAGE_TEMP_DIR"
+    echo '{ }' > package.json
+    yarn add "$PACKAGE_NAME@$PACKAGE_VERSION" >> $LOG_FILE 2>&1
+    yarn add patch-package <> $LOG_FILE 2>&1
+
+    # Apply the patch with --merge option
+    patch -p1 --no-backup-if-mismatch --input="$PATCH_PATH" --forward --merge 2>&1 >> $LOG_FILE || true
```
Comment (odelcroi) [inline scripts/merge-patches.sh:47]: I received this error on MacOs with version patch 2.5.8
`patch: unrecognized option `--merge'`

File: scripts/merge-patches.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+set -e
+
+# Variables
+PROJECT_DIR=$(pwd)
+PACKAGE_JSON="$PROJECT_DIR/package.json"
+PATCHES_DIR="$PROJECT_DIR/patches"
+TEMP_DIR="$PROJECT_DIR/temp"
+
+function merge_patches() {
```
Comment (odelcroi) [inline scripts/merge-patches.sh:11]: it would be cool to be able to launch only on one patch by its name

Comment (jdauphant) [reply scripts/merge-patches.sh:11]: we can refactor, ChatGPT is able to do that

File: scripts/merge-patches.sh (hunk 3 above)
Comment (odelcroi) [reply scripts/merge-patches.sh:47]: use docker

File: patches/simplify-exchange-key-message/matrix-react-sdk+3.68.0.patch [hunk 5]
```diff
@@ -39,32 +39,32 @@ index 68cb728..78e3894 100644
              <div>
                  <Spinner />
 +                {/* :TCHAP: remove code
-                 <p>{_t("Waiting for partner to confirm...")}</p>
+                 <p>{_t("Waiting for partner to confirm…")}</p>
```
Comment (odelcroi) [inline patches/simplify-exchange-key-message/matrix-react-sdk+3.68.0.patch:42]: c'est quoi ce truc?

**Pull request merged**